src/lib/common/selector.py

```diff
@@ -34,15 +36,22 @@ def load(self):
         """ the select DF is loaded from the appropriate file """
         return pd.read_csv(self.CSV, encoding="utf-8")
 
-    def index_complete(self, df, logs):
-        """ the data (list of lists) is saved in the appropriate file """
-        # TODO: do checks to ensure a valid structure, information rich enough.
+    def index(self, config):
+        df = self.index_files(config)
         if df is not None:
             df.to_csv(self.SELECT_MAP)
-        save_logs(logs, self.SELECT_LOGS)
+        save_logs(self.__selectLogs, self.SELECT_LOGS)
+
+    def select_logger(self, msg):
+        self.__selectLogs.append(msg)
+        print(msg)
+
+    def retrieve_logger(self, msg):
+        self.__retrieveLogs.append(msg)
+        print(msg)
 
     @abstractmethod
-    def index(self, config):
+    def index_files(self, config):
```

Comment:
Is there a reason you changed the name here to `index_files`? I think `index` makes more sense, as it is not necessarily files that a selector indexes: it's elements, i.e. videos on Youtube.

Reply:
Changed method name back to `index`, and renamed the method called on the selector from run.py `start_indexing`